scipy/optimize/_zeros_py.py

```diff
@@ -1388,3 +1388,278 @@ def toms748(f, a, b, args=(), k=1,
+    iqi = np.zeros(shape, dtype=bool)
+
+    funcalls = np.full(np.shape(flag), 0)
+    terminate = False
+
+    while iterations.any() < maxiter:
+        # use t to linearly interpolate between a and b,
+        # and evaluate this function as our newest estimate xt
+        xt = a + t * (b - a)
+        ft = f(xt, *args)
+        funcalls += 1
+
+        # update our history of the last few points so that
+        # - a is the newest estimate (we're going to update it from xt)
+        # - c and b get the preceding two estimates
+        # - a and b maintain opposite signs for f(a) and f(b)
+        samesign = np.sign(ft) == np.sign(fa)
+        c = np.choose(samesign, [b, a])
+        b = np.choose(samesign, [a, b])
+        fc = np.choose(samesign, [fb, fa])
+        fb = np.choose(samesign, [fa, fb])
+        a = xt
+        fa = ft
+
+        # set r so that f(r) is the minimum magnitude of f(a) and f(b)
+        fa_is_smaller = np.abs(fa) < np.abs(fb)
+        r = np.choose(fa_is_smaller, [b, a])
+        fm = np.choose(fa_is_smaller, [fb, fa])
+
+        tol = 2 * xatol * np.abs(r) + xrtol
+        tlim = tol / np.abs(b - c)
+        terminate = np.logical_or(terminate,
+                                  np.logical_or(fm == 0, tlim > 0.5))
+
+        # check convergence here, update flag if so
+        if np.all(terminate):
+            flag.fill(_ECONVERGED)
+            break
+
+        iterations += 1 - terminate
+
+        # Figure out values xi and phi
+        # to determine which method we should use next
+        xi = (a - b) / (c - b)
+        phi = (fa - fb) / (fc - fb)
+        iqi = np.logical_and(phi**2 < xi, (1 - phi) ** 2 < 1 - xi)
+
+        if iqi.all():
+            t = np.full(shape, 0.5)
+            a2, b2, c2, fa2, fb2, fc2 = (
+                a[iqi],
+                b[iqi],
+                c[iqi],
+                fa[iqi],
+                fb[iqi],
+                fc[iqi],
+            )
+            t[iqi] = fa2 / (fb2 - fa2) * fc2 / (fb2 - fc2) + (c2 - a2) / (
+                b2 - a2
+            ) * fa2 / (fc2 - fa2) * fb2 / (fc2 - fb2)
+        else:
+            # bisection
+            t = 0.5
+
+        # limit to the range (tlim, 1-tlim)
+        t = np.minimum(1 - tlim, np.maximum(tlim, t))
+
+        intermediate_result = OptimizeResult(x=r, fun=ft)
+        if _call_callback_maybe_halt(callback, intermediate_result):
+            break
+
+    # reshape outputs
+    r = np.reshape(r, intermediate_shape)
+    ft = np.reshape(ft, intermediate_shape)
+    funcalls = np.reshape(funcalls, intermediate_shape)
+    a = np.reshape(a, intermediate_shape)
+    b = np.reshape(b, intermediate_shape)
+    fa = np.reshape(fa, intermediate_shape)
+    fb = np.reshape(fb, intermediate_shape)
+    iterations = np.reshape(iterations, intermediate_shape)
+
+    # checks for convergence and conditions
+    if iterations.any() >= maxiter:
+        flag.fill(_ECONVERR)
```

Comment:
This also doesn't look right. Why would we fill _all_ the flags with `_ECONVERR` if _any_ element of iterations exceeds `maxiter`?

Reply:
Good question - I think I assumed that was the case, but let me try to fix that.